src/networking.c

```diff
@@ -3914,7 +4203,26 @@ void clientKillCommand(client *c) {
      * only after we queued the reply to its output buffers. */
     if (close_this_client) c->flag.close_after_reply = 1;
 client_kill_done:
-    zfree(client_filter.ids);
+    freeClientFilter(&client_filter);
+}
+
+static void freeClientFilter(clientFilter *filter) {
+    if (filter->ids != NULL)
+        zfree(filter->ids);
+    if (filter->flags != NULL)
+        sdsfree(filter->flags);
+    if (filter->capa != NULL)
+        sdsfree(filter->capa);
+    if (filter->ip != NULL)
+        sdsfree(filter->ip);
```

Comment:
```suggestion
    zfree(filter->ids);
    sdsfree(filter->flags);
    sdsfree(filter->capa);
    sdsfree(filter->ip);
```

All of these functions have built in null checks.

Reply:
I think it's safe to do a NULL check before free-ing. I agree we should `NULL` it out.

Reply:
Added them.